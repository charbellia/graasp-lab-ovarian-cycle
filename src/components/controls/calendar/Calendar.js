import React from 'react';
import PropTypes from 'prop-types';
// creating an svg file and pass
// the ovarian cycle dayCount to display
const Calendar = ({ dayCount, themeColor }) => (
  <svg
    aria-hidden="true"
    data-prefix="far"
    data-con="calendar"
    className="svg-inline--fa fa-calendar fa-w-14"
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 448 512"
  >
    <g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <g id="calendar" fill={themeColor}>
        <g id="calendar-regular" fillRule="nonzero">
          <path
            d="M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z"
            id="Shape"
          />
        </g>
        <text fill="black" id="date" fontFamily="Helvetica" fontSize="160" fontWeight="normal">
          <tspan x="168" y="356">{dayCount}</tspan>
        </text>
      </g>
    </g>
  </svg>
);

Calendar.propTypes = {
  dayCount: PropTypes.number.isRequired,
  themeColor: PropTypes.string.isRequired,
};

export default (Calendar);
