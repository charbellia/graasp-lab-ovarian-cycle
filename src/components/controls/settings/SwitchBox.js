import React from 'react';
import PropTypes from 'prop-types';
import { CirclePicker } from 'react-color';
import { Col, Row } from 'reactstrap';
import { withTranslation } from 'react-i18next';

const SwitchBox = ({
  color,
  handleChangeComplete,
  t,
}) => (
  <div className="control-container">
    <Row>
      <Col xs="12">
        <h4 className="modal-color-title">{t('Choose your panel color')}</h4>
        <CirclePicker
          color={color}
          onChangeComplete={handleChangeComplete}
        />
      </Col>
    </Row>
  </div>
);

SwitchBox.propTypes = {
  t: PropTypes.func.isRequired,
  handleChangeComplete: PropTypes.func.isRequired,
  color: PropTypes.string.isRequired,
};

export default withTranslation()(SwitchBox);
