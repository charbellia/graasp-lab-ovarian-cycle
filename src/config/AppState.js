import {
  COMMON_DOTS_0,
  COMMON_DOTS_1,
  COMMON_DOTS_2,
  COMMON_DOTS_3,
  COMMON_DOTS_4,
  COMMON_DOTS_5,
  COMMON_DOTS_6,
  COMMON_DOTS_7,
  COMMON_DOTS_8,
  COMMON_DOTS_9,
  COMMON_DOTS_10,
  MAX_LH,
} from './constants';

export const AppState = {
  postOvulation: false,
  preOvulation: true,
  ovulation: false,
  ovulationActive: false,
  postOvulationActive: false,
  preOvulationActive: false,
  isStarted: false,
  // we delay five seconds during the 13 and 14 days
  delay: 5,
  dayCount: 1, // we initialize the day counter to 1
  secretLhFsh: false, // at the begening we do not secret any hormones
  secretProgest: false,
  secretOestro: false,
  // starting state for the progesterone dots
  progesteronePoints: [
    [378, 1070],
    COMMON_DOTS_0,
    COMMON_DOTS_1,
    COMMON_DOTS_2,
    COMMON_DOTS_3,
    COMMON_DOTS_4,
    [382, 400],
    [383, 260],
    [383, 200],
  ],
  // ending state for the progesterone dots
  // starting state for the oestrogene dots
  oestrogenePoints: [
    [370, MAX_LH],
    COMMON_DOTS_0,
    COMMON_DOTS_1,
    COMMON_DOTS_2,
    COMMON_DOTS_3,
    COMMON_DOTS_4,
    [470, 400],
    [472, 260],
    [470, 200],
  ],
  // ending state for the oestrogene dots
  // starting state for the lh dots
  lhPoints: [
    [458, 200],
    [458, 260],
    [448, 400],
    COMMON_DOTS_5,
    COMMON_DOTS_6,
    COMMON_DOTS_7,
    COMMON_DOTS_8,
    COMMON_DOTS_9,
    COMMON_DOTS_10,
    [485, 1070],
    [482, MAX_LH],
  ],
  // ending state for the oestrogene dots
  // starting state for the fsh dots
  fshPoints: [
    [395, 210],
    [394, 260],
    [404, 400],
    COMMON_DOTS_5,
    COMMON_DOTS_6,
    COMMON_DOTS_7,
    COMMON_DOTS_8,
    COMMON_DOTS_9,
    COMMON_DOTS_10,
    [370, MAX_LH],
  ],
  // ending state for the fsh dots
};

export default AppState;
