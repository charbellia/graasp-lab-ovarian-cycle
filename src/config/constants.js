export const MAX_LH = 1085;
export const SKY_BLUE = '#12a3c1';
export const GREEN = '#3bc71f';
export const ORANGE = '#ff7a00';
export const PURPLE_BLUE = '#9C27B0';
export const COMMON_DOTS = [414, 1020];
export const COMMON_DOTS_1 = [415, 930];
export const COMMON_DOTS_2 = [418, 830];
export const COMMON_DOTS_3 = [417, 700];
export const COMMON_DOTS_4 = [413, 470];
