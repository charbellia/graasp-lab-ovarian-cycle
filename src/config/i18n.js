import i18n from 'i18next';
import { reactI18nextModule } from 'react-i18next';
import en from '../locale/en';
import fr from '../locale/fr';
// this is our main internationalization file
// configuration for the react-i18next
// we return english as default by passing 'en' params
// to the fallbackLng function
// all our translation files are in the ./locale directory
i18n
  .use(reactI18nextModule)
  .init({
    resources: {
      en,
      fr,
    },
    fallbackLng: 'en',
    debug: true,
    ns: ['translations'],
    defaultNS: 'translations',
    keySeparator: false,
    interpolation: {
      escapeValue: false,
      formatSeparator: ',',
    },
    react: {
      wait: true,
    },
  });
export default i18n;
