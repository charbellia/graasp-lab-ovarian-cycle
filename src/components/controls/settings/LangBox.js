import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withNamespaces } from 'react-i18next';
import { Col, Row } from 'reactstrap';
import { connect } from 'react-redux';
import Select from 'react-select';
import { options as langOptions } from '../../../config/options';
import { changeLanguage } from '../../../actions';

class LangBox extends Component {
  static propTypes = {
    dispatchChangeLanguage: PropTypes.func.isRequired,
    t: PropTypes.func.isRequired,
    selectedLanguage: PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string,
    }).isRequired,
    i18n: PropTypes.shape({
      defaultNS: PropTypes.string,
    }).isRequired,
  };

  handleChangeLanguage = (lang) => {
    const newLang = lang.value;
    const { dispatchChangeLanguage, i18n } = this.props;
    i18n.changeLanguage(newLang);
    dispatchChangeLanguage(newLang);
  };

  render() {
    const {
      t,
      selectedLanguage,
    } = this.props;

    return (
      <div className="control-container">
        <Row className="modal-lang-container">
          <Col xs="4">
            <h4 className="modal-lang-title">
              { t('Language') }
            </h4>
          </Col>
          <Col xs="8">
            <Select
              value={selectedLanguage}
              defaultValue={langOptions[0]}
              options={langOptions}
              onChange={this.handleChangeLanguage}
              className="select-lang"
            />
          </Col>
        </Row>
      </div>
    );
  }
}

const mapStateToProps = (state, { i18n }) => {
  const { language } = i18n;
  const selectedLanguage = langOptions.find(langOption => langOption.value === language);
  return {
    selectedLanguage,
  };
};

const mapDispatchToProps = {
  dispatchChangeLanguage: changeLanguage,
};

const ConnectedComponent = connect(mapStateToProps, mapDispatchToProps)(LangBox);

export default withNamespaces()(ConnectedComponent);
