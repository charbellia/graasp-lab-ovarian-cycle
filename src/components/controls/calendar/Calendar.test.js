import React from 'react';
import { shallow } from 'enzyme';
import { Calendar } from './Calendar';
import { DEFAULT_THEME_COLOR } from '../../../config/settings';

describe('<Calendar />', () => {
  const props = {
    dayCount: 0,
    themeColor: DEFAULT_THEME_COLOR,
    classes: {},
  };

  const component = shallow(<Calendar {...props} />);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('has one <svg /> with class fa-calendar', () => {
    expect(component.find('svg.fa-calendar').length).toEqual(1);
  });
});
