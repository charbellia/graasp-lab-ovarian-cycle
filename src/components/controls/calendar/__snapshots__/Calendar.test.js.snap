// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calendar
    classes={Object {}}
    dayCount={0}
    themeColor="#5050d2"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-hidden": "true",
      "children": <g
        fill="#5050d2"
        fillRule="evenodd"
        stroke="none"
        strokeWidth="1"
      >
        <path
          d="M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z"
          fillRule="nonzero"
        />
        <text
          fill="#5050d2"
          fontFamily="Helvetica"
          fontSize="160"
          fontWeight="normal"
        >
          <tspan
            x="141"
            y="356"
          >
            00
          </tspan>
        </text>
      </g>,
      "className": "undefined rootCalendar svg-inline--fa fa-calendar fa-w-14",
      "data-con": "calendar",
      "data-prefix": "far",
      "role": "img",
      "viewBox": "0 0 448 512",
      "xmlns": "http://www.w3.org/2000/svg",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <path
            d="M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z"
            fillRule="nonzero"
          />,
          <text
            fill="#5050d2"
            fontFamily="Helvetica"
            fontSize="160"
            fontWeight="normal"
          >
            <tspan
              x="141"
              y="356"
            >
              00
            </tspan>
          </text>,
        ],
        "fill": "#5050d2",
        "fillRule": "evenodd",
        "stroke": "none",
        "strokeWidth": "1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z",
            "fillRule": "nonzero",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tspan
              x="141"
              y="356"
            >
              00
            </tspan>,
            "fill": "#5050d2",
            "fontFamily": "Helvetica",
            "fontSize": "160",
            "fontWeight": "normal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "00",
              "x": "141",
              "y": "356",
            },
            "ref": null,
            "rendered": "00",
            "type": "tspan",
          },
          "type": "text",
        },
      ],
      "type": "g",
    },
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-hidden": "true",
        "children": <g
          fill="#5050d2"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1"
        >
          <path
            d="M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z"
            fillRule="nonzero"
          />
          <text
            fill="#5050d2"
            fontFamily="Helvetica"
            fontSize="160"
            fontWeight="normal"
          >
            <tspan
              x="141"
              y="356"
            >
              00
            </tspan>
          </text>
        </g>,
        "className": "undefined rootCalendar svg-inline--fa fa-calendar fa-w-14",
        "data-con": "calendar",
        "data-prefix": "far",
        "role": "img",
        "viewBox": "0 0 448 512",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z"
              fillRule="nonzero"
            />,
            <text
              fill="#5050d2"
              fontFamily="Helvetica"
              fontSize="160"
              fontWeight="normal"
            >
              <tspan
                x="141"
                y="356"
              >
                00
              </tspan>
            </text>,
          ],
          "fill": "#5050d2",
          "fillRule": "evenodd",
          "stroke": "none",
          "strokeWidth": "1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M400,64 L352,64 L352,12 C352,5.4 346.6,0 340,0 L300,0 C293.4,0 288,5.4 288,12 L288,64 L160,64 L160,12 C160,5.4 154.6,0 148,0 L108,0 C101.4,0 96,5.4 96,12 L96,64 L48,64 C21.5,64 0,85.5 0,112 L0,464 C0,490.5 21.5,512 48,512 L400,512 C426.5,512 448,490.5 448,464 L448,112 C448,85.5 426.5,64 400,64 Z M394,464 L54,464 C50.7,464 48,461.3 48,458 L48,160 L400,160 L400,458 C400,461.3 397.3,464 394,464 Z",
              "fillRule": "nonzero",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tspan
                x="141"
                y="356"
              >
                00
              </tspan>,
              "fill": "#5050d2",
              "fontFamily": "Helvetica",
              "fontSize": "160",
              "fontWeight": "normal",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "00",
                "x": "141",
                "y": "356",
              },
              "ref": null,
              "rendered": "00",
              "type": "tspan",
            },
            "type": "text",
          },
        ],
        "type": "g",
      },
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
